File: libs/monobitmap/src/main/kotlin/mono/graphics/bitmap/drawable/CharDrawable.kt
package mono.graphics.bitmap.drawable

import mono.graphics.bitmap.MonoBitmap

/**
 * A drawable which simplify fills with [char].
 */
class CharDrawable(private val char: Char) : Drawable {
    override fun toBitmap(width: Int, height: Int): MonoBitmap {
        val builder = MonoBitmap.Builder(width, height)
        builder.fill(char)
        return builder.toBitmap()
    }
}

File: monosketch-svelte/src/lib/mono/monobitmap/drawable/char-drawable.ts
import { MonoBitmap } from "$mono/monobitmap/bitmap/monobitmap";
import type { Drawable } from "$mono/monobitmap/drawable/drawable";

/**
 * A drawable which simplify fills with [char].
 */
export class CharDrawable implements Drawable {
    constructor(private readonly char: string) {
    }

    toBitmap(width: number, height: number): MonoBitmap.Bitmap {
        const builder = new MonoBitmap.Builder(width, height);
        builder.fillAll(this.char);
        return builder.toBitmap();
    }
}